libs/ledger-live-common/src/device/hooks/useLatestFirmware.ts

```diff
@@ -0,0 +1,21 @@
+import { getProviderId } from "../../manager/index";
+import { getEnv } from "@ledgerhq/live-env";
+import { version } from "../../../package.json";
+import { HttpManagerApiRepository } from "../../device-core/managerApi/repositories/HttpManagerApiRepository";
+import { useGetLatestFirmware } from "../../device-react/hooks/useGetLatestFirmware";
+import { DeviceInfoEntity } from "../../device-core/managerApi/entities/DeviceInfoEntity";
+
+export function useLatestFirmware(
+  deviceInfo?: DeviceInfoEntity | null,
+  managerApiRepository: HttpManagerApiRepository = new HttpManagerApiRepository(
```

Comment:
[ASK] Is it safe to create several instances of the manager? This means we will have concurrent api calls and concurrent cache access?

Reply:
Great point, and I just noticed that the code in `HttpManagerAPIRepository` is very very bad regarding cache management, as every function call creates a new cache 😬